akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileAndResourceDirectives.scala

```diff
@@ -222,7 +222,16 @@ object FileAndResourceDirectives extends FileAndResourceDirectives {
             ResourceFile(url, entry.getSize, entry.getTime)
           }
         } finally jar.close()
-      case _ ⇒ None
+      case _ ⇒
+        val conn = url.openConnection()
+        try {
+          conn.setUseCaches(false) // otherwise the JDK will keep the JAR file open when we close!
+          val len = conn.getContentLength
+          val lm = conn.getLastModified
+          Some(ResourceFile(url, len, lm))
+        } catch {
+          case _: Throwable ⇒ None
```

Comment:
I was referring to catching `Throwable` here, which wasn't there before and is considered bad practice.

The example above is basically equivalent to what we had before #17495 but additionally filters out directories (comment added).

If you can verify that it works as is then all is well.